src/include/network/lan/address.rb

```diff
@@ -632,93 +629,72 @@ def InitSlave(_key)
       # reorder the items
       l2 = []
       l1 = []
-      Builtins.foreach(
-        Convert.convert(items, from: "list", to: "list <term>")
-      ) do |t|
-        if Builtins.contains(
-          Ops.get_list(@settings, "SLAVES", []),
-          Ops.get_string(t, [0, 0], "")
-          )
-          l1 = Builtins.add(l1, t)
+      items.each do |t|
```

Comment:
Maybe you could use [Array#partition](http://ruby-doc.org/core-2.1.9/Enumerable.html#method-i-partition) in this case.